projects/ad9172/src/main.c

```diff
@@ -0,0 +1,257 @@
+	struct axi_jesd204_tx *tx_jesd;
+	struct adxcvr *tx_adxcvr;
+	struct axi_dac *tx_dac;
+	int32_t status;
+
+	status = hmc7044_init(&hmc7044_device, &hmc7044_param);
+	if (status != SUCCESS) {
+		printf("hmc7044_init() error: %"PRIi32"\n", status);
+		goto error_1;
+	}
+
+	status = axi_jesd204_tx_init(&tx_jesd, &tx_jesd_init);
+	if (status != SUCCESS) {
+		printf("error: %s: axi_jesd204_rx_init() failed\n", tx_jesd_init.name);
+		goto error_2;
+	}
+
+	status = adxcvr_init(&tx_adxcvr, &tx_adxcvr_init);
+	if (status != SUCCESS) {
+		printf("error: %s: adxcvr_init() failed\n", tx_adxcvr_init.name);
+		goto error_3;
+	}
+
+	status = adxcvr_clk_enable(tx_adxcvr);
+	if (status != SUCCESS) {
+		printf("error: %s: adxcvr_clk_enable() failed\n", tx_adxcvr->name);
+		goto error_3;
+	}
+
+	status = axi_jesd204_tx_lane_clk_enable(tx_jesd);
+	if (status != SUCCESS) {
+		printf("error: %s: axi_jesd204_tx_lane_clk_enable() failed\n", tx_jesd->name);
+		goto error_3;
+	}
+
+	status = ad9172_init(&ad9172_device, &ad9172_param);
+	if (status != SUCCESS) {
+		printf("ad9172_init() error: %"PRIi32"\n", status);
+		goto error_4;
+	}
+
+	status = axi_jesd204_tx_status_read(tx_jesd);
+	if (status != SUCCESS) {
+		printf("axi_jesd204_tx_status_read() error: %"PRIi32"\n", status);
+		goto error_4;
+	}
+
+	status = axi_dac_init(&tx_dac, &tx_dac_init);
+	if (status != SUCCESS) {
+		printf("axi_jesd204_tx_status_read() error: %"PRIi32"\n", status);
+		goto error_5;
+	}
+
+	printf("Set dds frequency at 40MHz\n");
+
+	axi_dac_dds_set_frequency(tx_dac, 0, 40000000);	/* TX1_I_F1 */
+	axi_dac_dds_set_frequency(tx_dac, 1, 40000000);	/* TX1_I_F2 */
+	axi_dac_dds_set_frequency(tx_dac, 2, 40000000);	/* TX1_Q_F1 */
+	axi_dac_dds_set_frequency(tx_dac, 3, 40000000);	/* TX1_Q_F2 */
+
+	axi_dac_dds_set_frequency(tx_dac, 4, 40000000); /* TX2_I_F1 */
+	axi_dac_dds_set_frequency(tx_dac, 5, 40000000); /* TX2_I_F2 */
+	axi_dac_dds_set_frequency(tx_dac, 6, 40000000); /* TX2_Q_F1 */
+	axi_dac_dds_set_frequency(tx_dac, 7, 40000000); /* TX2_Q_F2 */
+
+
+	axi_dac_dds_set_scale(tx_dac, 0, 250000);
+	axi_dac_dds_set_scale(tx_dac, 1, 250000);
+	axi_dac_dds_set_scale(tx_dac, 2, 250000);
+	axi_dac_dds_set_scale(tx_dac, 3, 250000);
+
+	axi_dac_dds_set_scale(tx_dac, 4, 250000);
+	axi_dac_dds_set_scale(tx_dac, 5, 250000);
+	axi_dac_dds_set_scale(tx_dac, 6, 250000);
+	axi_dac_dds_set_scale(tx_dac, 7, 250000);
+
+
+	axi_dac_dds_set_phase(tx_dac, 0, 90000);
+	axi_dac_dds_set_phase(tx_dac, 1, 90000);
+	axi_dac_dds_set_phase(tx_dac, 2, 0);
+	axi_dac_dds_set_phase(tx_dac, 3, 0);
+
+	axi_dac_dds_set_phase(tx_dac, 4, 90000);
+	axi_dac_dds_set_phase(tx_dac, 5, 90000);
+	axi_dac_dds_set_phase(tx_dac, 6, 0);
+	axi_dac_dds_set_phase(tx_dac, 7, 0);
+
+	axi_dac_set_datasel(tx_dac, -1, AXI_DAC_DATA_SEL_DDS);
+
+	printf("Bye\n");
+
+	axi_dac_remove(tx_dac);
+
+	ad9172_remove(ad9172_device);
+
+	adxcvr_remove(tx_adxcvr);
+
+	axi_jesd204_tx_remove(tx_jesd);
+
+	hmc7044_remove(hmc7044_device);
```

Comment:
I would delete all these *_remove() functions and leave only the ones with the error labels bellow. At the end of the main function, you can return 0.